Index: includes/QueryPage.php
===================================================================
--- includes/QueryPage.php	(revision 3749)
+++ includes/QueryPage.php	(working copy)
@@ -396,6 +396,13 @@
 			# $res might contain the whole 1,000 rows, so we read up to
 			# $num [should update this to use a Pager]
 			for( $i = 0; $i < $num && $row = $dbr->fetchObject( $res ); $i++ ) {
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+				$title = Title::makeTitleSafe( $row->namespace, $row->title );
+				if (!$title->userCanReadEx()) {
+					continue;
+				}
+/*op-patch|TS|2009-06-19|end*/  
+				
 				$line = $this->formatResult( $skin, $row );
 				if( $line ) {
 					$attr = ( isset( $row->usepatrol ) && $row->usepatrol && $row->patrolled == 0 )
