Index: includes/specials/SpecialListredirects.php
===================================================================
--- includes/specials/SpecialListredirects.php	(revision 3749)
+++ includes/specials/SpecialListredirects.php	(working copy)
@@ -38,6 +38,12 @@
 		if( $revision ) {
 			# Make a link to the destination page
 			$target = Title::newFromRedirect( $revision->getText() );
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+			if (!$target->userCanReadEx()) {
+				return;
+			}
+/*op-patch|TS|2009-06-19|end*/  
+			
 			if( $target ) {
 				$arr = $wgContLang->getArrow() . $wgContLang->getDirMark();
 				$targetLink = $skin->makeLinkObj( $target );
