Index: includes/specials/SpecialSearch.php
===================================================================
--- includes/specials/SpecialSearch.php	(revision 3749)
+++ includes/specials/SpecialSearch.php	(working copy)
@@ -340,7 +340,14 @@
 		$out .= "<ul class='mw-search-results'>\n";
 
 		while( $result = $matches->next() ) {
-			$out .= $this->showHit( $result, $terms );
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+			if (($result->getTitle() != NULL) 
+			    && ($result->getTitle()->userCanReadEx())) {
+/*op-patch|TS|2009-06-19|end*/  
+				$out .= $this->showHit( $result, $terms );
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+			}
+/*op-patch|TS|2009-06-19|end*/  
 		}
 		$out .= "</ul>\n";
 
