Index: includes/specials/SpecialRecentchanges.php
===================================================================
--- includes/specials/SpecialRecentchanges.php	(revision 3749)
+++ includes/specials/SpecialRecentchanges.php	(working copy)
@@ -352,6 +352,14 @@
 		$dbr = wfGetDB( DB_SLAVE );
 
 		foreach( $rows as $obj ){
+
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+			$rc = RecentChange::newFromRow( $obj );
+			if (!$rc->getTitle()->userCanReadEx()) {
+				continue;
+			}
+/*op-patch|TS|2009-06-19|end*/  
+			
 			if( $limit == 0) {
 				break;
 			}
