Index: includes/specials/SpecialWatchlist.php
===================================================================
--- includes/specials/SpecialWatchlist.php	(revision 3749)
+++ includes/specials/SpecialWatchlist.php	(working copy)
@@ -290,6 +290,12 @@
 	while ( $obj = $dbr->fetchObject( $res ) ) {
 		# Make RC entry
 		$rc = RecentChange::newFromRow( $obj );
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+		if (!$rc->getTitle()->userCanReadEx()) {
+			continue;
+		}
+/*op-patch|TS|2009-06-19|end*/  
+		
 		$rc->counter = $counter++;
 
 		if ( $wgShowUpdatedMarker ) {
