Index: includes/specials/SpecialPrefixindex.php
===================================================================
--- includes/specials/SpecialPrefixindex.php	(revision 3749)
+++ includes/specials/SpecialPrefixindex.php	(working copy)
@@ -102,6 +102,11 @@
 	
 				while( ($n < $this->maxPerPage) && ($s = $dbr->fetchObject( $res )) ) {
 					$t = Title::makeTitle( $s->page_namespace, $s->page_title );
+/*op-patch|TS|2009-06-19|HaloACL|SafeTitle|start*/
+					if ($t && !$t->userCanReadEx()) {
+						continue; 
+					}
+/*op-patch|TS|2009-06-19|end*/  
 					if( $t ) {
 						$link = ($s->page_is_redirect ? '<div class="allpagesredirect">' : '' ) .
 							$sk->makeKnownLinkObj( $t, htmlspecialchars( $t->getText() ), false, false ) .
