Index: extensions/SemanticMediaWiki/includes/SMW_QueryProcessor.php
===================================================================
--- extensions/SemanticMediaWiki/includes/SMW_QueryProcessor.php	(revision 8537)
+++ extensions/SemanticMediaWiki/includes/SMW_QueryProcessor.php	(working copy)
@@ -268,46 +268,62 @@
 		return $result;
 	}
 
-	static public function getResultFromQuery( $query, array $params, $extraprintouts, $outputmode, $context = SMWQueryProcessor::INLINE_QUERY, $format = '' ) {
-		wfProfileIn( 'SMWQueryProcessor::getResultFromQuery (SMW)' );
-		
-		// Query routing allows extensions to provide alternative stores as data sources
-		// The while feature is experimental and is not properly integrated with most of SMW's architecture. For instance, some query printers just fetch their own store.
-		///TODO: case-insensitive
-		global $smwgQuerySources;
-		
-		if ( array_key_exists( "source", $params ) && array_key_exists( $params["source"], $smwgQuerySources ) ) {
-			$store = new $smwgQuerySources[$params["source"]]();
-			$query->params = $params; // this is a hack
-		} else {
-			$store = smwfGetStore(); // default store
-		}
-		
-		$res = $store->getQueryResult( $query );
+    static public function getResultFromQuery( $query, array $params, $extraprintouts, $outputmode, $context = SMWQueryProcessor::INLINE_QUERY, $format = '' ) {
+        wfProfileIn( 'SMWQueryProcessor::getResultFromQuery (SMW)' );
 
-		if ( ( $query->querymode == SMWQuery::MODE_INSTANCES ) || ( $query->querymode == SMWQuery::MODE_NONE ) ) {
-			wfProfileIn( 'SMWQueryProcessor::getResultFromQuery-printout (SMW)' );
-			
-			if ( $format == '' ) {
-				$format = SMWQueryProcessor::getResultFormat( $params );
-			}
-			
-			$printer = SMWQueryProcessor::getResultPrinter( $format, $context, $res );
-			$result = $printer->getResult( $res, $params, $outputmode );
-			
-			wfProfileOut( 'SMWQueryProcessor::getResultFromQuery-printout (SMW)' );
-			wfProfileOut( 'SMWQueryProcessor::getResultFromQuery (SMW)' );
-			
-			return $result;
-		} else { // result for counting or debugging is just a string
-			if ( array_key_exists( 'intro', $params ) )  $res = str_replace( '_', ' ', $params['intro'] ) . $res;
-			if ( array_key_exists( 'outro', $params ) )  $res .= str_replace( '_', ' ', $params['outro'] );
-			
-			wfProfileOut( 'SMWQueryProcessor::getResultFromQuery (SMW)' );
-			
-			return $res . smwfEncodeMessages( $query->getErrors() );
-		}
-	}
+        // Query routing allows extensions to provide alternative stores as data sources
+        // The while feature is experimental and is not properly integrated with most of SMW's architecture. For instance, some query printers just fetch their own store.
+        ///TODO: case-insensitive
+        global $smwgQuerySources;
+
+        if ( array_key_exists( "source", $params ) && array_key_exists( $params["source"], $smwgQuerySources ) ) {
+            $store = new $smwgQuerySources[$params["source"]]();
+            $query->params = $params; // this is a hack
+        } else {
+            $store = smwfGetStore(); // default store
+        }
+
+        $resultHTML = "";
+        $res = $store->getQueryResult( $query );
+
+        if (!is_array($res)) {
+            $qResults['tsc'] = $res;
+        } else {
+            $qResults = $res;
+        }
+    
+        foreach($qResults as $source => $res) {
+
+            if ($source != 'tsc') {
+                $resultHTML .= "\n==$source==\n";
+            }
+            
+            if ( ( $query->querymode == SMWQuery::MODE_INSTANCES ) || ( $query->querymode == SMWQuery::MODE_NONE ) ) {
+                wfProfileIn( 'SMWQueryProcessor::getResultFromQuery-printout (SMW)' );
+
+                if ( $format == '' ) {
+                    $format = SMWQueryProcessor::getResultFormat( $params );
+                }
+
+                $printer = SMWQueryProcessor::getResultPrinter( $format, $context, $res );
+                $result = $printer->getResult( $res, $params, $outputmode );
+
+                wfProfileOut( 'SMWQueryProcessor::getResultFromQuery-printout (SMW)' );
+                wfProfileOut( 'SMWQueryProcessor::getResultFromQuery (SMW)' );
+
+                $resultHTML .= $result;
+            } else { // result for counting or debugging is just a string
+                if ( array_key_exists( 'intro', $params ) )  $res = str_replace( '_', ' ', $params['intro'] ) . $res;
+                if ( array_key_exists( 'outro', $params ) )  $res .= str_replace( '_', ' ', $params['outro'] );
+
+                wfProfileOut( 'SMWQueryProcessor::getResultFromQuery (SMW)' );
+
+                $resultHTML .= $res . smwfEncodeMessages( $query->getErrors() );
+            }
+
+        }
+        return $resultHTML;
+    }
 
 	/**
 	 * Determines the format from an array of parameters, and returns it.
